editor/editor_help.cpp

```diff
@@ -2581,19 +2581,19 @@ static void _add_text_to_rt(const String &p_bbcode, RichTextLabel *p_rt, Control
 			const String codeblock_text = bbcode.substr(brk_end + 1, end_pos - (brk_end + 1)).strip_edges();
 
 			String codeblock_copy_text = codeblock_text;
-			if (using_tab_indent) {
-				// Replace the code block's space indentation with tabs.
+			if (!using_tab_indent) {
+				// Replace the code block's tab indentation with spaces.
 				StringBuilder builder;
 				PackedStringArray text_lines = codeblock_copy_text.split("\n");
 				for (const String &line : text_lines) {
 					const String stripped_line = line.dedent();
-					const int space_count = line.length() - stripped_line.length();
+					const int tab_count = line.length() - stripped_line.length();
 
 					if (builder.num_strings_appended() > 0) {
 						builder.append("\n");
 					}
-					if (space_count > 0) {
-						builder.append(String("\t").repeat(MAX(space_count / 4, 1)) + stripped_line);
+					if (tab_count > 0) {
+						builder.append(String(" ").repeat(MAX(tab_count * 4, 1)) + stripped_line);
```

Comment:
We can use the editor setting `text_editor/behavior/indent/size` instead of hardcode 4.

```suggestion
						builder.append(String(" ").repeat(tab_count * indent_size) + stripped_line);
```

```c++
const int indent_size = MAX(1, int(EDITOR_GET("text_editor/behavior/indent/size")));
```